velox/functions/prestosql/tests/UuidFunctionsTest.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+#include "velox/functions/prestosql/types/UuidType.h"
+
+namespace facebook::velox::functions::prestosql {
+
+namespace {
+
+class UuidFunctionsTest : public functions::test::FunctionBaseTest {};
+
+TEST_F(UuidFunctionsTest, uuid) {
+  auto result =
+      evaluate<FlatVector<int128_t>>("uuid()", makeRowVector(ROW({}), 10));
+
+  // Sanity check results. All values are unique.
+  std::unordered_set<int128_t> uuids;
+  for (auto i = 0; i < 10; ++i) {
+    const auto uuid = result->valueAt(i);
+    ASSERT_TRUE(uuids.insert(uuid).second);
+  }
+  ASSERT_EQ(10, uuids.size());
+}
+
+TEST_F(UuidFunctionsTest, castAsVarchar) {
+  auto result = evaluate<FlatVector<StringView>>(
+      "cast(uuid() as varchar)", makeRowVector(ROW({}), 10));
+
+  // Sanity check results. All strings are unique. Each string is 36 bytes
+  // long.
+  std::unordered_set<std::string> uuids;
+  for (auto i = 0; i < 10; ++i) {
+    const auto uuid = result->valueAt(i).str();
+    ASSERT_EQ(36, uuid.size());
+    ASSERT_TRUE(uuids.insert(uuid).second);
+  }
+  ASSERT_EQ(10, uuids.size());
+}
+
+TEST_F(UuidFunctionsTest, castRoundTrip) {
```

Comment:
We can add a test where casting from UUID to type other than varchar throws the exception.

Reply:
Good idea. Adding

```
TEST_F(UuidFunctionsTest, unsupportedCast) {
  auto input = makeRowVector(ROW({}), 10);
  VELOX_ASSERT_THROW(
      evaluate("cast(uuid() as integer)", input),
      "Cannot cast UUID to INTEGER");
  VELOX_ASSERT_THROW(
      evaluate("cast(123 as uuid())", input), "Cannot cast BIGINT to UUID.");
}
```